### ISSUE TITLE
AttributeError: 'super' object has no attribute 'response'

### ISSUE THREAD

**Issue by mseymour** [Flexget/Flexget#890]
After upgrading Flexget from an old version (sometime late last year), I have been getting these errors when using any command. I have also not been able to access HTTPS URLs either, and that was before (and after) upgrading.

```
2016-02-26 20:18 CRITICAL plugin                        Exception while loading plugin flexget.plugins.api.history
2016-02-26 20:18 ERROR    plugin                        'super' object has no attribute 'response'
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/flexget/plugin.py", line 381, in _load_plugins_from_dirs
    __import__(module_name)
  File "/Library/Python/2.7/site-packages/flexget/plugins/api/history.py", line 9, in <module>
    from flexget.api import api, APIResource
  File "/Library/Python/2.7/site-packages/flexget/api/__init__.py", line 310, in <module>
    loader.find_module(module_name).load_module(module_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/pkgutil.py", line 246, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/Library/Python/2.7/site-packages/flexget/api/authentication.py", line 81, in <module>
    class LoginAPI(APIResource):
  File "/Library/Python/2.7/site-packages/flexget/api/authentication.py", line 83, in LoginAPI
    @api.response(401, 'Invalid username or password')
  File "/Library/Python/2.7/site-packages/flexget/api/__init__.py", line 147, in response
    return super(Api, self).response(code_or_apierror, description, model=model)
AttributeError: 'super' object has no attribute 'response'
Traceback (most recent call last):
  File "/usr/local/bin/flexget", line 9, in <module>
    load_entry_point('FlexGet==1.2.460', 'console_scripts', 'flexget')()
  File "/Library/Python/2.7/site-packages/flexget/__init__.py", line 37, in main
    manager.start()
  File "/Library/Python/2.7/site-packages/flexget/manager.py", line 299, in start
    self.initialize()
  File "/Library/Python/2.7/site-packages/flexget/manager.py", line 173, in initialize
    plugin.load_plugins(extra_dirs=[os.path.join(self.config_base, 'plugins')])
  File "/Library/Python/2.7/site-packages/flexget/plugin.py", line 425, in load_plugins
    _load_plugins_from_dirs(extra_dirs)
  File "/Library/Python/2.7/site-packages/flexget/plugin.py", line 381, in _load_plugins_from_dirs
    __import__(module_name)
  File "/Library/Python/2.7/site-packages/flexget/plugins/api/history.py", line 9, in <module>
    from flexget.api import api, APIResource
  File "/Library/Python/2.7/site-packages/flexget/api/__init__.py", line 310, in <module>
    loader.find_module(module_name).load_module(module_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/pkgutil.py", line 246, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/Library/Python/2.7/site-packages/flexget/api/authentication.py", line 81, in <module>
    class LoginAPI(APIResource):
  File "/Library/Python/2.7/site-packages/flexget/api/authentication.py", line 83, in LoginAPI
    @api.response(401, 'Invalid username or password')
  File "/Library/Python/2.7/site-packages/flexget/api/__init__.py", line 147, in response
    return super(Api, self).response(code_or_apierror, description, model=model)
AttributeError: 'super' object has no attribute 'response'
```

**Comment by gazpachoking** [comment]
It feels as though this must be related to [this thing](https://landscape.io/github/Flexget/Flexget/310/modules/flexget/api/__init__.py#L134) that landscape has been trying to tell us.

**Comment by stevezau** [comment]
This might be an issue with the version of restplus you have installed...

How did you upgrade flexget? Via pip??

Can you run the below and paste the output

```pip show flask-restplus```

**Comment by mseymour** [comment]
Yes, on OS X 10.11.

`sudo -H pip install --install-option '--install-data=/usr/local' --ignore-installed six --upgrade flexget`

And the output of `pip show flash-restplus`:

```
Hollis:~ mark$ pip show flask-restplus
---
Metadata-Version: 1.1
Name: flask-restplus
Version: 0.7.2
Summary: Helpers, syntaxic sugar and Swagger documentation for Flask-Restful
Home-page: https://github.com/noirbizarre/flask-restplus
Author: Axel Haustant
Author-email: axel@data.gouv.fr
License: MIT
Location: /Library/Python/2.7/site-packages/flask_restplus-0.7.2-py2.7.egg
Requires: flask-restful
```

**Comment by stevezau** [comment]
that's the issue. For some reason it's using an older version of flask-restplus..

We require 0.8.6 (https://github.com/Flexget/Flexget/blob/develop/pavement.py#L55)

I'm not sure why it didn't upgrade that package..

You can manually fix by pip install flask-restplus==0.8.6

**Comment by mseymour** [comment]
Looks like it decided to install the old version regardless?

```
Collecting flask-restplus==0.8.6
  Downloading flask_restplus-0.8.6-py2.py3-none-any.whl (1.1MB)
    100% |████████████████████████████████| 1.1MB 162kB/s
Requirement already satisfied (use --upgrade to upgrade): aniso8601>=0.82 in /Library/Python/2.7/site-packages/aniso8601-1.1.0-py2.7.egg (from flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): pytz in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): jsonschema in /Library/Python/2.7/site-packages/jsonschema-2.5.1-py2.7.egg (from flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): flask-restful>=0.3.2 in /Library/Python/2.7/site-packages/Flask_RESTful-0.3.5-py2.7.egg (from flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): python-dateutil in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from aniso8601>=0.82->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): functools32 in /Library/Python/2.7/site-packages (from jsonschema->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): Flask>=0.8 in /Library/Python/2.7/site-packages/Flask-0.10.1-py2.7.egg (from flask-restful>=0.3.2->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): six>=1.3.0 in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from flask-restful>=0.3.2->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): Werkzeug>=0.7 in /Library/Python/2.7/site-packages/Werkzeug-0.11.2-py2.7.egg (from Flask>=0.8->flask-restful>=0.3.2->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): Jinja2>=2.4 in /Library/Python/2.7/site-packages/Jinja2-2.8-py2.7.egg (from Flask>=0.8->flask-restful>=0.3.2->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): itsdangerous>=0.21 in /Library/Python/2.7/site-packages/itsdangerous-0.24-py2.7.egg (from Flask>=0.8->flask-restful>=0.3.2->flask-restplus==0.8.6)
Requirement already satisfied (use --upgrade to upgrade): MarkupSafe in /Library/Python/2.7/site-packages (from Jinja2>=2.4->Flask>=0.8->flask-restful>=0.3.2->flask-restplus==0.8.6)
Installing collected packages: flask-restplus
  Found existing installation: flask-restplus 0.7.2
    Uninstalling flask-restplus-0.7.2:
      Successfully uninstalled flask-restplus-0.7.2
Successfully installed flask-restplus-0.7.2
```

**Comment by gazpachoking** [comment]
Hmm, maybe make sure pip and setuptools are fully upgraded and try again?

**Comment by stevezau** [comment]
that's weird.. maybe try uninstall first?  pip uninstall flask-restplus

**Comment by mseymour** [comment]
I already upgraded pip to 8.0.3 and setuptools to 1.1.6. Using `--force-reinstall` made pip install 0.8.6.

This also fixed flexget, although it doesn't fix `[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:590)` for me (which I would think would require a separate issue.)

**Comment by stevezau** [comment]
yeah open another issue for that one.
